core/src/main/java/com/linecorp/armeria/client/endpoint/WeightRampingUpStrategyBuilder.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.endpoint;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+import java.util.concurrent.ScheduledExecutorService;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.CommonPools;
+
+import io.netty.channel.EventLoop;
+
+/**
+ * Builds a weight ramping up {@link EndpointSelectionStrategy} which ramps the weight of newly added
+ * {@link Endpoint}s. The {@link Endpoint} is selected using weighted random distribution.
+ */
+public final class WeightRampingUpStrategyBuilder {
+
+    private static final long DEFAULT_RAMPING_UP_INTERVAL_MILLIS = 2000;
+    private static final int DEFAULT_NUMBER_OF_STEPS = 10;
+    private static final int DEFAULT_RAMPING_UP_ENTRY_WINDOW_MILLIS = 500;
+
+    private EndpointWeightTransition transition = EndpointWeightTransition.linear();
+
+    @Nullable
+    private ScheduledExecutorService executor;
+
+    private long rampingUpIntervalMillis = DEFAULT_RAMPING_UP_INTERVAL_MILLIS;
+    private int totalSteps = DEFAULT_NUMBER_OF_STEPS;
+    private long rampingUpTaskWindowMillis = DEFAULT_RAMPING_UP_ENTRY_WINDOW_MILLIS;
+
+    /**
+     * Sets computing the weight of an {@link Endpoint} during the transition with the specified
+     * {@link EndpointWeightTransition}. {@link EndpointWeightTransition#linear()} is used by default.
+     */
+    public WeightRampingUpStrategyBuilder transition(EndpointWeightTransition transition) {
+        this.transition = requireNonNull(transition, "transition");
+        return this;
+    }
+
+    /**
+     * Sets the specified {@link ScheduledExecutorService} that will be used to compute the weight of
```

Comment:
Sets the ScheduledExecutorService to use to execute tasks for computing new weights.

By the way is it sake for this to be multi threaded or should it accept a netty type instead?

Reply:
Oh, it should be single-threaded. Let me use `EventExecutor`. Thanks!